### PR TITLE
becoming a skeleton will now cure your husking

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Upon having your species changed to the skeleton species, any forms of husking you may have endured will be cured.

## Why It's Good For The Game

According to the comments and code in https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/carbon/human/status_procs.dm , skeletons are not intended to ever be husks. If something attempts to husk a skeleton, the cure_husk() proc is called to cure any existing huskings and the become_husk() proc returns out early.

However, if a husked corpse with a stage 4+ necrotic metabolism + metabolic boost virus is allowed to process a dosage of skeleton mutation toxin for long enough to turn into a skeleton, they can be turned into a skeleton while still a husk, and this will not cure their husking. A similar thing can occur if a ghoul or a voiceless dead (with or without the aforementioned virus) is allowed to process a dosage of skeleton mutation toxin for long enough to turn into a skeleton. There are probably also some pride mirror or summon events shenanigans you can pull to become a husked skeleton.

This PR patches the above extreme edge cases by making it so that if you get turned into a skeleton for any reason, your husking(s) will be cured.

**tldr;
skeleton no husk except in edge case. pr fix edge case. pr good.**

## Changelog
:cl: ATHATH
fix: You can no longer become a husked skeleton by becoming a skeleton after being husked.
/:cl:

### CODE REVIEW

**Review by coiax — CHANGES_REQUESTED** [review]
Good intent, I'd prefer a slightly different execution.

Create a TRAIT_HUSK_IMMUNE, scoped to `/mob/living/carbon/human` (as in, leave a comment that the trait is only for that type, and is meaningless anywhere else).

Put a signal listener on humans when TRAIT_HUSK_IMMUNE is gained to call `cure_husk()`.

Instead of the species type check on skeletons, just check for the trait instead.

And then make sure that the edge case bug you're fixing is still fixed.

**Comment by ATH1909** [comment]
> Good intent, I'd prefer a slightly different execution.
> 
> Create a TRAIT_HUSK_IMMUNE, scoped to `/mob/living/carbon/human` (as in, leave a comment that the trait is only for that type, and is meaningless anywhere else).
> 
> Put a signal listener on humans when TRAIT_HUSK_IMMUNE is gained to call `cure_husk()`.
> 
> Instead of the species type check on skeletons, just check for the trait instead.
> 
> And then make sure that the edge case bug you're fixing is still fixed.

but that's effort

**Comment by Mothblocks** [comment]
image
